packages/client/src/features/mouse-tool/mouse-tool.ts

```diff
@@ -85,15 +71,13 @@ export class RankingMouseTool extends MouseTool implements IMouseTool {
         }
     }
 
-    async dispatchActions<K extends keyof MouseListener>(
+    async dispatchActions<K extends MouseListenerMethods>(
         mouseListeners: MouseListener[],
         methodName: K,
         element: SModelElement,
         event: MouseEvent
     ): Promise<void> {
-        const actions = mouseListeners
-            .map(listener => listener[methodName].apply(listener, [element, event]))
-            .reduce((a, b) => a.concat(b));
+        const actions = mouseListeners.map(listener => listener[methodName](element, event as WheelEvent)).reduce((a, b) => a.concat(b));
```

Comment:
Could you explain why the event is cast to `WheelEvent`? I believe this method is generic for any MouseEvent but the cast also does not seem to break anything. In any case, if you want to leave it, please explain why this is happening.

Reply:
Sure: the sprotty `MouseListener` as a `wheel` method which expects a `WheelEvent` as parameter. 
So due to new strict typescript configuration the compiler would complain if  we wouldn't cast the event. 
However casting is totally safe in this case because `WheelEvent` extends `MouseEvent` and can therefore also be safely used in all other listener methods that just expect a `MouseEvent`.
The original sprotty implementation handles this in the same fashion: 
https://github.com/eclipse/sprotty/blob/4c111d2a96b8e2bfef6acb290ec0d185ddb752c7/packages/sprotty/src/base/views/mouse-tool.ts#L66

Reply:
I see, thank you for the explanation. I saw that it was not causing problems but still it looked odd. If sprotty does it in the same way we can definitely do the same.